### ISSUE TITLE
Move into X11::XCB namespace

### ISSUE THREAD

**Issue by slobo** [skids/perl6-xcb#1]
Hey skids,
Just had a chance to look through what you have so far. It's looking good, but now I see what you meant by namespace collision ;) I think it would make sense to have this package under `X11::XCB` rather than annexing top level  `X11`.
Let me know what you think and if you need help with that. I think once moved in there and adding some automated build integration (eg travis) we may as well add it to ecosystem to increase visibility and gather comments.
Cheers

**Comment by skids** [comment]
Hi... I'm kind of torn on that.  The stuff already in X11::XCB definitely belongs there.  The stuff I currently have in X11:: is more a higher level abstraction of the X11 protocol... on the one hand it needs some of the guts of libxcb to work (at least, at present... a pure-perl core/event loop is not completely out of the realm of possibility)... on  the other hand it's not really a wrapper for ibxcb, just something new that builds on it.

There's a whole lot to do before this is ready for the ecosystem... or even to seriously suggest for use for anyone.  A lot more thought needs to be put into cleaning up the rather immense and, for historical reasons, cobbled-together namespace, and that's complicated further by the autogenning of code from XCBXML.  I've been way bogged down at work, but hope to get back to it eventually -- hpefully after it is ore or less feature complete I'll have a better picture on how to deal with the top-level namespace issue... it turns out you actually have t implement an application using it to fully appreciate how seemingly tiny decisions about namespacing affect usability :-)

**Comment by slobo** [comment]
I think you raise good points. When I started making X11::Xlib::Raw, one purpose was to learn about Xlib (hence doing bindings manually instead of trying gptrixie etc), and I had limited the scope to direct port of xlib semantics with minimal Perlishness sprinkled on top (the only changes so far are a couple of C macros that deal with structs that I put as methods on the struct directly). At the same time, I've been transliterating simple C/C++ programs into Perl just to see it all working. Perhaps once Xlib::Raw is complete I would start looking into creating a more natural perl interface to it....

X11::XCB seems to have ambitions set quite a bit higher, gunning for natural Perl 6 interface right away. Perhaps a good first step to get stuff into ecosystem is to carve out autogenerated XCB::Raw bits into own package that can be released, and you can then build the rest on top of it as a separate project? Not sure how tangled autogen code is with the rest of current implementation, but my hunch is that it could be useful on its own.

**Comment by skids** [comment]
Yeah, there are a few core features I want to get working first -- transferring file descriptors being the biggest.  Then after that I have to eliminate the last of the things that the autogen code skips and ensure we generate relatively compatible code from both prevelent versions of XMLXCB (the current head, and the older one that most distros seem to ship).

And after that, it's all down to untangling the mess -- figuring out what the best solution is to the various namespace problems internally within the autogen code because I've been a bit too heavy handed already (and that'll be a real pain with the input extensions), and better separation of concerns between the XCB core + autogen and the more high level stuff.  Then finally figuring out how to reconcile the versioning.  That's when I expect to make a final call on the top level namespace.
